sourmash/sbtmh.py

```diff
@@ -77,6 +85,210 @@ def data(self, new_data):
+                        i not in self._leaves,
+                        i not in self._missing_nodes)):
+                    next_internal_node = i
+                    break
+        if next_internal_node is None:
+            next_node = max(self._leaves.keys()) + 1
+        else:
+            next_node = next_internal_node
+
+        return next_node
+
+    def add_node(self, node):
+        pos = self.new_node_pos(node)
+
+        if pos == 0:  # empty tree; initialize w/node.
+            new_internal_node = Node(self.factory, name="internal." + str(pos))
+            self._nodes[0] = new_internal_node
+            pos = self.new_node_pos(node)
+
+        # Cases:
+        # 1) parent is a Leaf (already covered)
+        # 2) parent is a Node (with empty position available)
+        #    - add Leaf, update parent
+        # 3) parent is a Node (no position available)
+        #    - this is covered by case 1
+        # 4) parent is None
+        #    this can happen with d != 2, in this case create the parent node
+        parent = self.parent(pos)
+        if isinstance(parent.node, Leaf):
+            # Create a new internal node
+            # node and parent are children of new internal node
+            # Move children of grandparent internal node to new inserted internal node
+            # below, and bump new node up to grandparent node
+            grandparent = self.parent(parent.pos)
+            grandparent_children = self.children(grandparent.pos)
+            if grandparent_children[0] == parent:
+                parent_sibling = grandparent_children[1]
+            else:
+                parent_sibling = grandparent_children[0]
+            if isinstance(parent_sibling.node, Node):
+                self.insert_new_internal_node_with_children(node, parent)
+            else:
+                self.relocate_children_to_new_internal_node(grandparent, node, parent,
+                                                            parent_sibling)
+        elif isinstance(parent.node, Node):
+            self._leaves[pos] = node
+            node.update(parent.node)
+        elif parent.node is None:
+            new_internal_node = Node(self.factory, name="internal." + str(parent.pos))
+            self._nodes[parent.pos] = new_internal_node
+            c1 = self.children(parent.pos)[0]
+            self._leaves[c1.pos] = node
+            node.update(new_internal_node)
+
+        # update all parents!
+        parent = self.parent(parent.pos)
+        while parent:
+            self._rebuild_node(parent.pos)
+            node.update(self._nodes[parent.pos])
+            parent = self.parent(parent.pos)
+
+    def relocate_children_to_new_internal_node(self, grandparent, node, parent,
+                                               parent_sibling):
+        new_internal_node = Node(self.factory, name="internal." + str(parent.pos))
+        self._nodes[parent.pos] = new_internal_node
+        c1, c2 = self.children(parent.pos)[:2]
```

Comment:
could you unpack this instead of index?

```suggestion
        c1, c2 = *self.children(parent.pos)
```